packages/connect-core/src/entities/Role.ts

```diff
@@ -1,45 +1,52 @@
 import CoreEntity from './CoreEntity'
 import Permission, { PermissionData } from './Permission'
 import { AragonArtifact } from '../types'
-import { parseMetadata } from '../utils/parseMetadata'
+import { resolveMetadata } from '../utils/metadata'
 import { ConnectorInterface } from '../connections/ConnectorInterface'
 
 export interface RoleData {
   appAddress: string
   artifact?: string | null
+  contentUri?: string | null
   hash: string
   manager?: string
   grantees?: PermissionData[] | null
 }
 
 export default class Role extends CoreEntity implements RoleData {
   readonly appAddress!: string
-  readonly description?: string
-  readonly permissions?: Permission[] | null
   readonly hash!: string
-  readonly name?: string
+  readonly permissions?: Permission[] | null
   readonly manager?: string
   readonly params?: string
+  #artifact?: string | null
+  #contentUri?: string
+  description?: string
+  name?: string
 
-  constructor({ artifact, ...data }: RoleData, connector: ConnectorInterface) {
+  constructor(data: RoleData, connector: ConnectorInterface) {
     super(connector)
 
-    // TODO: If no metadata, fallback to resolve ourselves with ipfs
-
-    if (artifact) {
-      const { roles }: AragonArtifact = parseMetadata(artifact, 'artifact.json')
-
-      const role = roles.find(role => role.bytes === data.hash)
-
-      this.name = role?.id
-      this.description = role?.name
-    }
-
     this.appAddress = data.appAddress
+    this.hash = data.hash
     this.permissions = data.grantees?.map(
       grantee => new Permission(grantee, connector)
     )
-    this.hash = data.hash
     this.manager = data.manager
+    this.#artifact = data.artifact
+    this.#contentUri = data.contentUri ?? undefined
```

Comment:
Same as above; should we return `null`?

Reply:
I’m thinking that the only reason to return `undefined` here would be for `contentUri` to be `undefined`, since `??` would exclude an empty string. Should we use `||` instead?

Reply:
Gotcha; yes this makes sense to me.